app/api/crud.py

```diff
@@ -177,28 +178,62 @@ async def get(
     return response_obj
 
 
-async def get_terms(data_element_URI: str) -> dict:
+async def get_terms(
+    data_element_URI: str, term_labels_path: Path | None
+) -> dict:
     """
     Makes a POST request to graph API using httpx where the payload is a SPARQL query generated by the create_terms_query function.
 
     Parameters
     ----------
     data_element_URI : str
         Controlled term of neurobagel class for which all the available terms should be retrieved.
+    term_labels_path : Path
+        Path to JSON file containing term-label mappings for the vocabulary of the data element URI.
 
     Returns
     -------
     dict
-        Dictionary where the key is the Neurobagel class and values correspond to all the terms representing available (i.e. used) instances of that class in the graph.
+        Dictionary where the key is the Neurobagel class and the value is a list of dictionaries
+        corresponding to the available (i.e. used) instances of that class in the graph. Each instance dictionary
+        has two items: the 'TermURL' and the human-readable 'Label' for the term.
     """
-    results = post_query_to_graph(util.create_terms_query(data_element_URI))
+    term_url_results = post_query_to_graph(
+        util.create_terms_query(data_element_URI)
+    )
+
+    if term_labels_path is not None:
+        vocab_term_lookup = util.load_json(term_labels_path)
+
+    term_label_dicts = []
+    for result in term_url_results["results"]["bindings"]:
+        term_url = result["termURL"]["value"]
+        # First, check whether the found instance of the data element contains a recognized namespace
+        if util.is_term_namespace_in_context(term_url):
+            # Then, check if we have a vocab lookup available to get the human-readable label for the term
+            if term_labels_path is not None:
+                # If so, attempt to get the label for the term
+                term_label = vocab_term_lookup.get(
+                    util.strip_namespace_from_term_uri(term_url), None
+                )
+            else:
+                term_label = None
```

Comment:
could we replace that by having an `else` statement above for when `term_labels_path is not None` and then just setting `vocab_term_lookup = {}`?